rest/src/main/scala/io/udash/rest/RestDataCompanion.scala

```diff
@@ -43,27 +43,18 @@ abstract class RestDataCompanion[T](implicit
   * It must be a singleton object type, i.e. `SomeObject.type`.
   */
 abstract class RestDataCompanionWithDeps[D, T](implicit
-  deps: ValueOf[D], instances: MacroInstances[(DefaultRestImplicits, D), CodecWithStructure[T]]
+  deps: ValueOf[D],
+  instances: MacroInstances[(DefaultRestImplicits, D), CodecWithStructure[T]],
 ) extends AbstractRestDataCompanion[(DefaultRestImplicits, D), T]((DefaultRestImplicits, deps.value))
 
 /**
-  * Base class for companion objects of wrappers over other data types (i.e. case classes with single field).
-  * This companion ensures instances of all the REST typeclasses (serialization, schema, etc.) for wrapping type
-  * assuming that these instances are available for the wrapped type.
-  *
-  * Using this base companion class makes the wrapper class effectively "transparent", i.e. as if it was annotated with
-  * [[com.avsystem.commons.serialization.transparent transparent]] annotation.
-  *
-  * @example
-  * {{{
-  *   case class UserId(id: String) extends AnyVal
-  *   object UserId extends RestDataWrapperCompanion[String, UserId]
-  * }}}
-  */
-abstract class RestDataWrapperCompanion[Wrapped, T](implicit
-  instances: MacroInstances[DefaultRestImplicits, () => NameAndAdjusters[T]]
-) extends TransparentWrapperCompanion[Wrapped, T] {
-  private def nameAndAdjusters: NameAndAdjusters[T] = instances(DefaultRestImplicits, this).apply()
+  * Base trait for companion objects of wrappers over other data types (i.e. case classes with single field).
+  * Ensures instances of all the REST typeclasses (serialization, schema, etc.) for wrapping type [[T]]
+  * assuming that these instances are available for the wrapped type [[Wrapped]].
+ */
+trait BaseRestDataWrapperCompanion[Wrapped, T] {
```

Comment:
Can you add an example (as close to your real use case) in tests? Maybe we could find a better name for what this is then. Also, nobody removes it as "refactoring".

Reply:
It turns out something like this:
```
abstract class MyCustomRestCompanion[T](
  implicit instances: MacroInstances[DefaultRestImplicits, () => NameAndAdjusters[T]]
) extends RestDataWrapperCompanion[String, T]
```
is actually working, so it turns out I don't need these changes :sweat_smile: